ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -459,32 +473,35 @@ - (BOOL) isTablet {
 - (float) getFontScale {
     // Font scales based on font sizes from https://developer.apple.com/ios/human-interface-guidelines/visual-design/typography/
     float fontScale = 1.0;
-    UITraitCollection *traitCollection = [[UIScreen mainScreen] traitCollection];
-
-    // Shared application is unavailable in an app extension.
-    if (traitCollection) {
-        __block NSString *contentSize = nil;
-        RCTUnsafeExecuteOnMainQueueSync(^{
-            if (@available(iOS 10.0, tvOS 10.0, macCatalyst 13.0, *)) {
-                contentSize = traitCollection.preferredContentSizeCategory;
-            } else {
-                // if we can't get contentSize, we'll fall back to 1.0
-            }
-        });
-
-        if ([contentSize isEqual: @"UICTContentSizeCategoryXS"]) fontScale = 0.82;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryS"]) fontScale = 0.88;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryM"]) fontScale = 0.95;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryL"]) fontScale = 1.0;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryXL"]) fontScale = 1.12;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryXXL"]) fontScale = 1.23;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryXXXL"]) fontScale = 1.35;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryAccessibilityM"]) fontScale = 1.64;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryAccessibilityL"]) fontScale = 1.95;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryAccessibilityXL"]) fontScale = 2.35;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryAccessibilityXXL"]) fontScale = 2.76;
-        else if ([contentSize isEqual: @"UICTContentSizeCategoryAccessibilityXXXL"]) fontScale = 3.12;
-    }
+
+    #if !TARGET_OS_VISION
+        UITraitCollection *traitCollection = [[UIScreen mainScreen] traitCollection];
```

Comment:
You can retrieve trait collection for visionOS using: `[UITraitCollection currentTraitCollection]` and have this part of code working for visionOS

Reply:
very interesting - @gupta-ji6 if you can get that in that would be great. Incremental improvement (that is: doing it as a follow-on) would also be fine - releases are automated here so I'd just merge a PR and release again via github action